src/common/function/src/scalars/timestamp/to_timezone.rs

```diff
@@ -0,0 +1,207 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use common_query::error::{InvalidFuncArgsSnafu, Result, UnsupportedInputDataTypeSnafu};
+use common_query::prelude::{Signature, Volatility};
+use common_time::timezone::TimeZone;
+use common_time::Timestamp;
+use datatypes::prelude::{ConcreteDataType, Vector};
+use datatypes::vectors::{StringVector, VectorRef};
+use snafu::ensure;
+
+use crate::scalars::function::{Function, FunctionContext};
+
+#[derive(Clone, Debug, Default)]
+pub struct ToTimeZoneFunction;
+
+const NAME: &str = "to_timezone";
+
+fn convert_to_timezone(arg: &str) -> Option<TimeZone> {
+    match TimeZone::from_tz_string(arg) {
+        Ok(ts) => ts,
+        Err(_err) => None,
+    }
+}
+
+fn convert_to_timestrap(arg: &str) -> Option<Timestamp> {
+    match Timestamp::from_str(arg) {
+        Ok(ts) => Some(ts),
+        Err(_) => None,
+    }
+}
+
+fn process_to_timezone(
+    times: Vec<Option<Timestamp>>,
+    tzs: Vec<Option<TimeZone>>,
+) -> Vec<Option<String>> {
+    times
+        .iter()
+        .zip(tzs.iter())
+        .map(|(time, tz)| match (time, tz) {
+            (Some(time), _) => Some(time.to_timezone_aware_string(tz.clone())),
+            _ => None,
+        })
+        .collect::<Vec<Option<String>>>()
+}
+
+impl Function for ToTimeZoneFunction {
+    fn name(&self) -> &str {
+        NAME
+    }
+
+    fn return_type(&self, _input_types: &[ConcreteDataType]) -> Result<ConcreteDataType> {
+        Ok(ConcreteDataType::string_datatype())
```

Comment:
I am not sure the return_type?

Reply:
https://github.com/GreptimeTeam/greptimedb/blob/5bd80a74abe1f16a769e6298aaad4d64347a6464/src/common/time/src/timestamp.rs#L34-L38

Does timestamp member `value` mean the `$value * $unit` after the ` 1 January 1970`??
If so, I think the value should be unchanged, and return the `time string` with `local time zone`.

Reply:
The timestamp is always the offset after `1970-1-1`. Timestamp itself isn't timezone aware so there is no need to convert a timestamp to a timestamp. I think this function only needs to format the timestamp with a specific timezone.

So it isn't as powerful as PostgresSQL's `timestamptz` type.
https://stackoverflow.com/questions/5876218/difference-between-timestamps-with-without-time-zone-in-postgresql
```sql
foo=> SET TIMEZONE TO 'Australia/Melbourne';
SET
foo=> SELECT '2011-01-01 00:00:00'::TIMESTAMP;
      timestamp      
---------------------
 2011-01-01 00:00:00
(1 row)
```

PostgreSQL also provides a `timezone()` function but it converts a `text` to a timestamp according to time zone. Actually.

A more flexible way might be to allow attaching the timezone info to our vector's data type like arrow:
https://docs.rs/arrow/latest/arrow/array/struct.PrimitiveArray.html#method.with_timezone